tools/afl-fuzzer/main.c

```diff
@@ -0,0 +1,476 @@
+
+	/*
+	 * sleep for 5 ms before continuing sending the next message.
+	 * This helps with the condition signaling. Without this,
+	 * the condition seems to always satisfy and
+	 * the fuzzer never waits for a response from the DSP.
+	 */
+	usleep(50000);
+
+	return ret;
+}
+
+int main(int argc, char *argv[])
+{
+	struct fuzz fuzzer;
+	int ret;
+	char *ipc_msg_file = NULL;
+	char *dsp_run_file = NULL;
+	char *platform_name = "byt";
+	char opt;
+	FILE *fp = NULL, *fp1 = NULL;
+
+	while ((opt = getopt(argc, argv, "m:d:")) != -1) {
+		switch (opt) {
+		case 'm':
+			ipc_msg_file = optarg;
+			break;
+		case 'd':
+			dsp_run_file = optarg;
+			break;
+		default:
+			exit(EXIT_FAILURE);
+		}
+	}
+
+	if (!ipc_msg_file) {
+		fprintf(stderr, "error: Need to provide ipc-msg file using -m option\n");
+		exit(EXIT_FAILURE);
+	}
+
+	if (!dsp_run_file) {
+		// TODO: Re-write error statement to make it clear
+		fprintf(stderr, "error: Need to provide dsp initializer file using -d option\n");
+		exit(EXIT_FAILURE);
+	}
+
+	fp1 = fopen(dsp_run_file, "r");
+
+	if (fp1 == NULL) {
+		fprintf(stderr, "error: Opening dsp initializer file %s\n",
+			dsp_run_file);
+		exit(EXIT_FAILURE);
+	}
+
+	int ch = fgetc(fp1);
+	int to_boot = ch;
+	fclose(fp1);
+
+	if (ch == EOF) {
+		fp1 = fopen(dsp_run_file, "w");
+
+		if (!fp1) {
+			fprintf(stderr, "error: opening dsp initializer file %s\n",
+				dsp_run_file);
+			exit(EXIT_FAILURE);
+		}
+
+		ch = 'y';
+		if (fputc(ch, fp1) != ch) {
+			fprintf(stderr, "error: writing to dsp initializer file %s\n",
+				dsp_run_file);
+			exit(EXIT_FAILURE);
+		}
+		fclose(fp1);
+	}
+
+	// Only run fork() if ch is 'y' else don't.
+	if (ch == 'y' && !fork()) {
+
+		char *newargv[] = { "/home/sof/work/qemu/build/xtensa-softmmu/qemu-system-xtensa",
+				    "-cpu", "baytrail", "-M", "adsp_byt", "-nographic", "-kernel",
+				    "/home/sof/work/sof.git/build_byt_gcc/src/arch/xtensa/sof-byt.ri",
+				    NULL};
+		char *newenviron[] = { NULL };
+
+		printf("Executing qemu xtensa\n");
+
+		execve(newargv[0], newargv, newenviron);
+
+		/* execve() returns only on error */
+		perror("execve");
+
+		exit(EXIT_FAILURE);
+
+	} else {
+
+		// TODO: Need to make sure that child is executed first.
+		// Need to sleep only if ch = 'y' as only then DSP is restarted.
+		if (ch == 'y') {
+			sleep(5);
+
+			// In the next iteration we needn't restart DSP
+			// unless there is FW crash
+			fp1 = fopen(dsp_run_file, "w");
+
+			if (!fp1) {
+				fprintf(stderr, "error: opening dsp initializer file %s\n",
+					dsp_run_file);
+				exit(EXIT_FAILURE);
+			}
+
+			ch = 'n';
+			if (fputc(ch, fp1) != ch) {
+				fprintf(stderr, "error: writing to dsp initializer file %s\n",
+					dsp_run_file);
+				exit(EXIT_FAILURE);
+			}
+
+			// TODO: Check error conditions for this.
+			fclose(fp1);
+		}
+
+
+
+		/* init platform */
+		fprintf(stdout, "initialising platform %s\n", platform[0]->name);
+		ret = platform[0]->init(&fuzzer, platform[0], to_boot);
```

Comment:
you hard coded the selected platform here, I think you mean to use the one passed in by the fuzzer struct

Reply:
Right now, working on adding support for platform byt. Instead of hard coding the selected platform, i updated the code so that the user gives the platform as input. (but the available platform for now is 'byt')